### PR TITLE
feat: impl Error for ClientCertVerifierBuilderError

### LINKED ISSUE #1454: impl `std::error::Error` for `ClientCertVerifierBuilderError`
`std::error::Error` is not implemented for [`ClientCertVerifierBuilderError`](https://docs.rs/rustls/0.22.0-alpha.2/rustls/server/enum.ClientCertVerifierBuilderError.html). It might be convenient for users who want to convert types between types which implement `std::error::Error` trait.

### PR DESCRIPTION
**Pull request by tottoto**
Resolves #1454.

### CODE REVIEW

File: rustls/src/webpki/client_verifier_builder.rs [hunk 1]
```diff
@@ -97,6 +98,17 @@ impl From<CertRevocationListError> for ClientCertVerifierBuilderError {
...
+impl fmt::Display for ClientCertVerifierBuilderError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::NoRootAnchors => write!(f, "no root trust anchors were provided"),
+            Self::InvalidCrl(e) => write!(f, "provided CRL could not be parsed: {:?}", e),
+        }
+    }
+}
+
+impl std::error::Error for ClientCertVerifierBuilderError {}
```
Comment (cpu) [inline rustls/src/webpki/client_verifier_builder.rs:110]: small nit: in other places in the codebase I see we tend to import `Error` and alias it to `StdError`:
https://github.com/rustls/rustls/blob/b08776c84d33a522e9f5c70a04ce01f5d5425669/rustls/src/sign.rs#L11

https://github.com/rustls/rustls/blob/b08776c84d33a522e9f5c70a04ce01f5d5425669/rustls/src/sign.rs#L407

Could you match that convention here?

**Review by cpu — APPROVED** [review]
One small nit, otherwise looks good to me. Thanks for reporting/fixing :-)

File: rustls/src/webpki/client_verifier_builder.rs (hunk 1 above)
Comment (tottoto) [reply rustls/src/webpki/client_verifier_builder.rs]: Fixed.

Comment (ctz) [reply rustls/src/webpki/client_verifier_builder.rs]: Thanks! Please squash that into the first commit.

Comment (tottoto) [reply rustls/src/webpki/client_verifier_builder.rs]: Done.

Comment (tottoto) [reply rustls/src/webpki/client_verifier_builder.rs]: Squached to one commit.

Comment (cpu) [reply rustls/src/webpki/client_verifier_builder.rs]: Thanks

**Comment by cpu** [comment]
@tottoto Would you mind rebasing on main? I can merge afterwards.

**Comment by tottoto** [comment]
Rebased.

**Pull request merged**